src/Stratis.Bitcoin.Features.Consensus.Tests/TimeSyncBehaviorTest.cs

```diff
@@ -28,173 +28,283 @@ public class TimeSyncBehaviorTest
+                TestSample.Inbound( true,  false, false,    0, TimeSpan.FromSeconds(-391),      new IPAddress(3)),                           
+
+                // These samples will change adjusted time because next outbound is the 4th outbound and we are under the limits.
+                TestSample.Outbound(true,  false, false, 1280, TimeSpan.FromSeconds(-1),        new IPAddress(4)),  // 2 inbound, 4 outbound. 2/4 * 3 = 1.5 -> ceil -> 2 of each outbound   { -2126000, -2126000, -391000, -1000, -1000, 3560, 3560, 13123, 26000, 26000 } -> median is 1280ms.
+                TestSample.Inbound( true,  false, false, 3560, TimeSpan.FromSeconds(23.6),      new IPAddress(5)),  // 3 inbound, 4 outbound. 3/4 * 3 = 2.25 -> ceil -> 3 of each outbound  { -2126000, -2126000, -2126000, -391000, -1000, -1000, -1000, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000 } -> median is 3560ms.                     
+                TestSample.Inbound( true,  false, false, 3560, TimeSpan.FromSeconds(236),       new IPAddress(6)),  // 4 inbound, 4 outbound. 4/4 * 3 = 3 -> ceil -> 3 of each outbound     { -2126000, -2126000, -2126000, -391000, -1000, -1000, -1000, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000, 236000 } -> median is 3560ms.                                                       
+                TestSample.Outbound(true,  false, false, 1236, TimeSpan.FromSeconds(1.236),     new IPAddress(7)),  // 4 inbound, 5 outbound. 4/5 * 3 = 2.4 -> ceil -> 3 of each outbound   { -2126000, -2126000, -2126000, -391000, -1000, -1000, -1000, 1236, 1236, 1236, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000, 236000 }  -> median is 1236ms.
+                TestSample.Inbound( true,  false, false, 1236, TimeSpan.FromSeconds(-1001),     new IPAddress(8)),  // 5 inbound, 5 outbound. 5/5 * 3 = 3 -> ceil -> 3 of each outbound     { -2126000, -2126000, -2126000, -391000, -1001, -1000, -1000, -1000, 1236, 1236, 1236, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000, 236000 }  -> median is 1236ms.
+                TestSample.Inbound( true,  false, false, 1236, TimeSpan.FromSeconds(-4.9236),   new IPAddress(9)),  // 6 inbound, 5 outbound. 6/5 * 3 = 3.6 -> ceil -> 4 of each outbound   { -2126000, -2126000, -2126000, -2126000, -391000, -4923.6, -1001, -1000, -1000, -1000, -1000, 1236, 1236, 1236, 1236, 3560, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000, 26000, 236000 }  -> median is 1236ms.
+                TestSample.Outbound(true,  false, false,  118, TimeSpan.FromSeconds(-4444.444), new IPAddress(10)), // 6 inbound, 6 outbound. 6/6 * 3 = 3 -> ceil -> 3 of each outbound    { -4444444, -4444444, -4444444, -2126000, -2126000, -2126000, -391000, -4923.6, -1001, -1000, -1000, -1000, 1236, 1236, 1236, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000, 236000 }  -> median is 118ms.
             };
+             
+            int maliciousOffset = TimeSyncBehaviorState.MaxTimeOffsetSeconds-1;
```

Comment:
always use `Ctrl+K + D` On your code once finished
This will improve the code style if you missed something, here it would be spaces around the minus

Reply:
normally do, will remember to in future.